frontend/lib/resolution/default-functions.cpp

```diff
@@ -509,6 +624,12 @@ getCompilerGeneratedMethodQuery(Context* context, const Type* type,
       result = generateDomainMethod(context, domainType, name);
     } else if (auto arrayType = type->toArrayType()) {
       result = generateArrayMethod(context, arrayType, name);
+    } else if (auto recordType = type->toRecordType()) {
+      if (name == USTR("==") || name == USTR("=")) {
```

Comment:
IMO it would be better to have a conditional here that checks these names and calls the appropriate helper based on the name, e.g.

```
if (name == USTR("==")) result = generateRecordAssignment(context, name, recordType);
else if (name == USTR("=")) result = generateRecordEquality(context, name, recordType);
else CHPL_ASSERT(false && "record method not implemented yet!");
```

(of course you probably should add `{ }` -- I just left those out for my comment here)